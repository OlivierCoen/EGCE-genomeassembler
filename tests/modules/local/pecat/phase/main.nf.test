nextflow_process {

    name "Test Process PECAT_PHASE"
    script "modules/local/pecat/phase/main.nf"
    process "PECAT_PHASE"
    tag "pecat_phase"

    test("Normal fasta") {

        tag "pecat_phase_normal"

        when {

            process {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true),
                    file( '$projectDir/tests/test_data/pecat/output/normal/first_assembly/results.tar.gz', checkIfExists: true)
                ]
                input[1] = file( '$projectDir/tests/test_data/pecat/input/pecat_optional_config.txt', checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Small fasta") {

        tag "pecat_phase_small"

        when {

            process {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true),
                    file( '$projectDir/tests/test_data/pecat/output/small/first_assembly/results.tar.gz', checkIfExists: true)
                ]
                input[1] = file( '$projectDir/tests/test_data/pecat/input/pecat_optional_config.txt', checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
