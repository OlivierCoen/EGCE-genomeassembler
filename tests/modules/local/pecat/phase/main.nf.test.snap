{
    "Small fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,141bf6f992ba79cdd5fcc8f58eafbf17"
                    ]
                ],
                "1": [
                    "versions.yml:md5,26d953ee02a19fe9345620d8796b0f36"
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,141bf6f992ba79cdd5fcc8f58eafbf17"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,26d953ee02a19fe9345620d8796b0f36"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-24T18:06:12.526469094"
    },
    "Normal fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,84ea207c16c94ced156f051f265020e3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,26d953ee02a19fe9345620d8796b0f36"
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,84ea207c16c94ced156f051f265020e3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,26d953ee02a19fe9345620d8796b0f36"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-24T20:25:07.677912372"
    }
}